mx.py

```diff
@@ -9567,17 +9575,36 @@ def processDep(dep, edge):
         if files:
             files.append(join(p.dir, '.factorypath'))
 
+def _get_ide_envvars():
+    """
+    Gets a dict of environment variables that must be captured in generated IDE configurations.
+    """
+    result = {'JAVA_HOME' : get_jdk().home}
+    names = [
+        # On the mac, applications are launched with a different path than command
+        # line tools, so capture the current PATH.  In general this ensures that
+        # the eclipse builders see the same path as a working command line build.
+        'PATH',
+        # The mx builders are run inside the directory of their associated suite,
+        # not the primary suite, so they might not see the env file of the primary
+        # suite.  Capture DEFAULT_VM in case it was only defined in the primary
+        # suite.
+        'DEFAULT_VM',
+        'MX_ALT_OUTPUT_ROOT',
+    ]
+    for name in names:
+        value = get_env(name)
+        if value is not None:
+            result[name] = value
+    return result
+
 def _capture_eclipse_settings(logToConsole):
     # Capture interesting settings which drive the output of the projects.
     # Changes to these values should cause regeneration of the project files.
-    value = 'logToConsole=%s\n' % logToConsole
-    if os.environ.get('PATH'):
-        value = value + 'PATH=%s\n' % os.environ['PATH']
-    if os.environ.get('JAVA_HOME'):
-        value = value + 'JAVA_HOME=%s\n' % os.environ['JAVA_HOME']
-    if os.environ.get('DEFAULT_VM'):
-        value = value + 'DEFAULT_VM=%s\n' % os.environ['DEFAULT_VM']
-    return value
+    settings = 'logToConsole=%s\n' % logToConsole
```

Comment:
@tkrodriguez , please check that this is semantically equivalent to how it was before my change.